lib/error_catalog.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef P4C_LIB_ERROR_CATALOG_H_
+#define P4C_LIB_ERROR_CATALOG_H_
+
+/// enumerate supported errors
+/// Backends should extend this class with additional errors in the range 500-999 and
+/// warnings in the range 1500-2141.
```

Comment:
wWhy is the range important? What happens if user decides to lower certain error to warning at compiler invocation time? Does the encoding move from range 500-999 to 1500-2141?

Reply:
No, the value of the enum won't move. The change in behavior is at the point of handling the warning/error, while the value just determines the format in the catalog.
The ranges are not essential, but I thought it would be good to partition the space for ease of further developing the notion of an error catalog.